### ISSUE TITLE
Why is there no confidence threshold for dinov3_eomt_semantic_segmentation predict

### ISSUE THREAD

**Issue by rabum** [lightly-ai/lightly-train#583]
As I see there is only one input:
<img width="1240" height="582" alt="Image" src="https://github.com/user-attachments/assets/39af25f6-c361-4e15-9123-c7b78f337602" />

Different from other models like dinov3_eomt_instance_segmentation

<img width="1212" height="666" alt="Image" src="https://github.com/user-attachments/assets/41571ff1-c706-4b3e-9e95-a5f823cfaec8" />

**Comment by stegmuel** [comment]
Hello,
It is because the nature of semantic segmentation is different from that of instance segmentation.
The objective is to classify every pixel in the image. The common practice is therefore to use an extra `background` class if there are pixels that do not belong to any of the original classes.

**Comment by rabum** [comment]
> Hello, It is because the nature of semantic segmentation is different from that of instance segmentation. The objective is to classify every pixel in the image. The common practice is therefore to use an extra `background` class if there are pixels that do not belong to any of the original classes.

My understanding is that, there should be a confidence threshold, and all pixels that do not meet the threshold should be classified as background

**Comment by stegmuel** [comment]
Both are valid approaches.
The advantage of the currently supported approach is that the model can learn the `background` representation.

**Comment by rabum** [comment]
> Both are valid approaches. The advantage of the currently supported approach is that the model can learn the `background` representation.

There might be a misunderstanding, I did not mean a different approach of training models, it's a post-process using threshold I was talking about

**Comment by stegmuel** [comment]
My point was that in the absence of a threshold in the `predict` method you can/should rely on an extra `background` class. 
Indeed, a threshold could also be used in the presence of `background` class. I'll see if we can add this.

**Comment by rabum** [comment]
> My point was that in the absence of a threshold in the `predict` method you can/should rely on an extra `background` class. Indeed, a threshold could also be used in the presence of `background` class. I'll see if we can add this.

Thanks, here is the change I made myself:
```
    def predict(
        self,
        image: PathLike | PILImage | Tensor,
        confidence_threshold: float | None = None,
        ignore_label: int = -1,
    ) -> Tensor:
        """Returns the predicted mask for the given image.

        Args:
            image:
                The input image as a path, URL, PIL image, or tensor. Tensors must have
                shape (C, H, W).

        Returns:
            The predicted mask as a tensor of shape (H, W). The values represent the
            class IDs as defined in the `classes` argument of your dataset. These
            classes are also stored in the `classes` attribute of the model.
            The model will always predict the pixels as one of the known classes even when
            your dataset contains ignored classes defined by the `ignore_classes` argument.
        """
        if self.training:
            self.eval()

        # Load image
        device = next(self.parameters()).device
        x = file_helpers.as_image_tensor(image).to(device)
        image_h, image_w = x.shape[-2:]

        x = transforms_functional.to_dtype(x, dtype=torch.float32, scale=True)
        x = transforms_functional.normalize(
            x, mean=self.image_normalize["mean"], std=self.image_normalize["std"]
        )
        # Crop size is the short side of the training image size. We resize the image
        # such that the short side of the image matches the crop size.
        crop_size = min(self.image_size)
        # (C, H, W) -> (C, H', W')
        x = transforms_functional.resize(x, size=[crop_size])
        x = x.unsqueeze(0)  # (1, C, H', W')

        logits = self._forward_logits(x)  # (1, K+1, H', W'), K = len(self.classes)
        # Restrict logits to known classes only.
        logits = logits[:, :-1]  # (1, K, H', W')
        logits = F.interpolate(
            logits, size=(image_h, image_w), mode="bilinear"
        )  # (1, K|K+1, H, W)

#        masks = logits.argmax(dim=1)  # (1, H, W)
#        # Map internal class IDs to class IDs.
#        masks = self.internal_class_to_class[masks]  # (1, H, W)
#        return masks[0]
        probs = torch.softmax(logits, dim=1)    # (1, K, H, W)
        max_probs, preds = probs.max(dim=1)     # (1, H, W)
    
        if confidence_threshold is not None:
            preds = preds.clone()
            preds[max_probs < confidence_threshold] = ignore_label

        preds = self.internal_class_to_class[preds]
    
        return preds[0]
```
